ignite/metrics/rouge.py

```diff
@@ -0,0 +1,189 @@
+
+    .. math::
+        F_\beta = \left( 1 + \beta^2 \right) * \frac{ \text{precision} * \text{recall} }
+        { \left( \beta^2 * \text{precision} \right) + \text{recall} }
+
+    where :math:`\beta` is a float.
+
+    - ``update`` must receive output of the form ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+    - `y_pred` must be a Sequence of Tokens.
+    - `y` must be in the form of List of Sequence of Tokens of Model Texts.
+
+    Args:
+        beta: beta value for calculating f-beta
+        metric: Which variant of Rouge Score to be evaluated
+            Valid Values - "rougeN", "rougen", "n", "N", "rougeL", "rougel", "l", "L"
+        version: specifically for Rouge-L to choose which flavour of Rouge-L you wish to use.
+            Valid Values - "sentence", "corpus"
+        aggregate: For multiple reference Rouge score calculation. Provides options to take mean or argmax of all
+            reference summaries for a given candidate summary.
+            Valid Values - "single", "mean", "max"
+        output_transform (callable, optional): a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+        device (Union[str, torch.device]) – specifies which device updates are accumulated on. Setting the metric’s
+            device to be the same as your update arguments ensures the update method is non-blocking. By default, CPU.
+
+    Example:
+
+    .. code-block:: python
+
+        from ignite.metrics import Rouge
+        m = Rouge(beta=1,metric="rouge-L",version="sentence",aggregate="single")
+        y_pred = "the cat was found under the bed"
+        y = "the cat was under the bed"
+        m.update([y_pred.split(),y.split()]]) #Using space to separate sentences into tokens
+        y_pred = "the tiny little cat was found under the big funny bed"
+        m.update([y_pred.split(),y.split()])
+        print(m.compute())
+
+    .. versionadded:: 0.5.0
+    """
+
+    def __init__(
+        self,
+        beta: float = float("inf"),
+        metric: str = "rouge-1",
+        aggregate: str = "single",
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ) -> None:
+        self._rougetotal: torch.Tensor = torch.tensor(0, device=device, dtype=torch.double)
+        self._num_examples: int = 0
+        self.beta: float = 0.0
+        self.n: int = 1
+        self.beta, self.n, self.rouge_fn, self.aggregate = self._check_parameters(beta, metric, aggregate)
+        self.alpha = 1 / (1 + self.beta ** 2)
+        super(Rouge, self).__init__(output_transform=output_transform, device=device)
+
+    def _check_parameters(self, beta: float, metric: str, aggregate: str) -> Tuple:
```

Comment:
```suggestion
    def _check_parameters(self, beta: float, metric: str, aggregate: str) -> Tuple[float, int, Callable, str]:
```
and thus we can possibly remove redefinitions:
```python
        self.beta: float = 0.0
        self.n: int = 1
```

Reply:
Fixed it!